go/nbs/table_persister.go

```diff
@@ -184,13 +187,13 @@ func (ftp fsTablePersister) Compact(mt *memTable, haver chunkReader) (name addr,
 		io.Copy(temp, bytes.NewReader(data))
 		return temp.Name(), name, chunkCount
 	}()
-	if chunkCount > 0 {
-		err := os.Rename(tempName, filepath.Join(ftp.dir, name.String()))
-		d.PanicIfError(err)
-	} else {
+	if chunkCount == 0 {
 		os.Remove(tempName)
+		return emptyChunkSource{}
 	}
-	return name, chunkCount
+	err := os.Rename(tempName, filepath.Join(ftp.dir, name.String()))
+	d.PanicIfError(err)
+	return newMmapTableReader(ftp.dir, name, chunkCount)
```

Comment:
it's a mild bummer that file table reader is still forced to read the bytes it just wrote, but kind of neccessary because of the need to setup the mmapping. one idea is if the table reader constructor took two byte slices, one for the prefix excluding the suffix index (since that is all eager converted to in-memory structures, and one for the suffixes, because that is left as lazily accessed memory.

in that case, the mmap guy would pass the data he's holding onto as well as correctly setup the mapping.

i'm find leaving that as an open bug. your call.